scc/src/opts.rs

```diff
@@ -50,85 +50,189 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 fn cache_dir() -> impl Parser<Option<PathBuf>> {
-    let tag = any::<String>("-customCacheDir").guard(|s| s == "-customCacheDir", "not customCacheDir");
-    let value = positional::<PathBuf>("CACHE_DIR");
-    construct!(tag, value).map(|pair| pair.1).optional().catch()
+    let tag = any::<String>("-customCacheDir")
+        .help("A custom cache dir to write the final.redscripts to")
+        .guard(|s| s == "-customCacheDir", "not customCacheDir");
+    let value = positional::<PathBuf>("CACHE_DIR")
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 impl Opts {
-    pub fn load(args: &[&str]) -> Self {
-        let optimize = toggle_options("optimize", "Optimize the redscripts");
-        let threads = slong("-threads", "Number of theads to script_paths on")
-            .parse(|s| s.parse::<u8>())
-            .fallback(1);
-        let no_warnings = toggle_options("Wnone", "No warnings");
-        let no_testonly = toggle_options("no-testonly", "No testonly classes");
-        let no_breakpoint = toggle_options("no-breakpoint", "No breakpoints");
-        let profile_off = toggle_options("profile=off", "Profile off");
-        let cache_file = any::<PathBuf>("CACHE_FILE").optional();
+    pub fn get_parser() -> OptionParser<Opts> {
+        let optimize = toggle_options("-optimize", "Optimize")
+            .parse(|s| Ok::<bool, String>(s.unwrap_or(Opts::default().optimize)));
+        let threads = slong("-threads", "THREADS", "Number of threads to compile on")
+            .parse(|s| s.map(|s| s.parse::<u8>()).unwrap_or(Ok(Opts::default().threads)));
+        let no_testonly = toggle_options("-no-testonly", "Exclude testonly classes")
+            .parse(|s| Ok::<bool, String>(s.unwrap_or(Opts::default().no_testonly)));
+        let no_breakpoint = toggle_options("-no-breakpoint", "Remove breakpoints")
+            .parse(|s| Ok::<bool, String>(s.unwrap_or(Opts::default().no_breakpoint)));
+        let warnings = no_space("-W", "Warning setting <none>");
+        let profile = equals_sign("-profile", "Profiling <yes|no>").parse(|s| {
+            if let Some(str) = s {
+                match str.as_str() {
+                    "on" => Ok(true),
+                    "off" => Ok(false),
+                    _ => Err("Profile option parse error"),
+                }
+            } else {
+                Ok(Opts::default().profile)
+            }
+        });
+        let cache_file = positional::<PathBuf>("CACHE_FILE")
+            .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -")
```

Comment:
same as some comments above:
```suggestion
        .guard(|s| matches!(s.to_str(), Some(str) if !str.starts_with('-')), "starts with -");
```
this is repeated many times, miht be a good idea to extract it to a function